### PR TITLE
refactor: TableNameKey and DatanodeTableKey

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The remaining two keys as discussed.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1728

### CODE REVIEW

**Review by WenyXu — APPROVED** [review]

File: src/common/meta/src/key.rs [hunk 1]
```diff
@@ -47,9 +63,25 @@ use crate::kv_backend::KvBackendRef;
 
 pub const REMOVED_PREFIX: &str = "__removed";
 
+const ALPHANUMERICS_NAME_PATTERN: &str = "[a-zA-Z_][a-zA-Z0-9_]*";
```
Comment (fengjiachun) [inline src/common/meta/src/key.rs:66]: TableName now can accept: "[a-zA-Z_:][a-zA-Z0-9_:]*"

https://github.com/GreptimeTeam/greptimedb/pull/1893/files#diff-c41f497328afde9ff1237b9105e5dcd0b2cc98401cf07a61ff1b4b550663adc3R33

**Comment by MichaelScofield** [comment]
@fengjiachun PTAL

File: src/common/meta/src/key.rs [hunk 2]
```diff
@@ -77,7 +77,7 @@ lazy_static! {
 
 lazy_static! {
     static ref TABLE_NAME_KEY_PATTERN: Regex = Regex::new(&format!(
-        "^{TABLE_NAME_KEY_PREFIX}/({ALPHANUMERICS_NAME_PATTERN})/({ALPHANUMERICS_NAME_PATTERN})/({ALPHANUMERICS_NAME_PATTERN})$"
+        "^{TABLE_NAME_KEY_PREFIX}/({TABLE_NAME_PATTERN})/({TABLE_NAME_PATTERN})/({TABLE_NAME_PATTERN})$"
```
Comment (fengjiachun) [inline src/common/meta/src/key.rs:80]: Do catalog/schema/table need their own patterns?

**Review by fengjiachun — APPROVED** [review]

File: src/common/meta/src/key.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/common/meta/src/key.rs:80]: I think we can keep it same with the table name pattern. It's naturally to think the naming restrictions - about catalog, schema and table - are equal.

**Pull request merged**